task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,309 @@
+              of source image pull specifications to their mirror pull specifications.
+
+              Args:
+                  imds_data (list): A list of dictionaries, where each dictionary
+                                    represents an ImageDigestMirrorSet object.
+                                    Expected structure:
+                                    [{'spec': {'imageDigestMirrors': [{'source': '...', 'mirrors': ['...']}, ...]}}, ...]
+
+              Returns:
+                  dict: A dictionary where keys are source image pull specifications
+                        (str) and values are lists of their corresponding mirror pull
+                        specifications (list of str). Returns an empty dictionary
+                        if no valid mirror sets are found or parsed.
+              """
+              result = {}
+              for imds in imds_data:
+                  if isinstance(imds, dict) and 'spec' in imds and 'imageDigestMirrors' in imds['spec']:
+                      for item in imds['spec']['imageDigestMirrors']:
+                          if 'source' in item and 'mirrors' in item:
+                              source_value = item['source']
+                              mirrors_list = item['mirrors']
+                              result[source_value] = mirrors_list
+                  else:
+                      print("Warning: Unexpected structure of an ImageDigestMirrorSet object "
+                            "(missing 'spec' or 'imageDigestMirrors'). Skipping this entry.")
+              return result
+
+
+          def get_correct_pullspec(mirror_set: dict, image_pullspec: str) -> str:
+              """
+              Determines the correct image pull specification by checking if the given
+              `image_pullspec` starts with any of the `source_pullspec` keys in the
+              `mirror_set`. If a match is found, it replaces the source part with
+              the first mirror in the corresponding list.
+
+              Args:
+                  mirror_set (dict): A dictionary mapping source pull specifications
+                                     to lists of mirror pull specifications.
+                                     (e.g., {'quay.io/old/repo': ['registry.example.com/new/repo']})
+                  image_pullspec (str): The original image pull specification to check.
+
+              Returns:
+                  str: The corrected image pull specification if a mirror is found,
+                       otherwise the original `image_pullspec` unchanged.
+              """
+              for source_pullspec in mirror_set.keys():
+                  if image_pullspec.startswith(source_pullspec):
+                      a = image_pullspec.replace(source_pullspec, mirror_set[source_pullspec][0])
+                      return a
+              return image_pullspec
+
+
+          def replace_pullspec(mirror_set: dict, catalog_data: list) -> list:
```

Comment:
Consider using an existing function? https://github.com/konflux-ci/konflux-test/blob/main/test/utils.sh#L838

Reply:
If you need to add new functionality, I would still do it in a common place so that we don't _just_ have it in this task.

Reply:
Functionality moved to konflux-test